hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -464,27 +464,43 @@ protected void postCommit(HoodieTable<T, I, K, O> table, HoodieCommitMetadata me
   }
 
   protected void runTableServicesInline(HoodieTable<T, I, K, O> table, HoodieCommitMetadata metadata, Option<Map<String, String>> extraMetadata) {
-    if (config.areAnyTableServicesInline()) {
+    if (config.areAnyTableServicesInline() || config.scheduleInlineTableServices()) {
       if (config.isMetadataTableEnabled()) {
         table.getHoodieView().sync();
       }
       // Do an inline compaction if enabled
       if (config.inlineCompactionEnabled()) {
         runAnyPendingCompactions(table);
         metadata.addMetadata(HoodieCompactionConfig.INLINE_COMPACT.key(), "true");
-        inlineCompact(extraMetadata);
+        inlineScheduleCompactAndOptionallyExecute(extraMetadata, !config.scheduleInlineCompaction());
       } else {
         metadata.addMetadata(HoodieCompactionConfig.INLINE_COMPACT.key(), "false");
       }
 
+      // if just inline schedule is enabled
+      if (!config.inlineCompactionEnabled() && config.scheduleInlineCompaction()
+          && !table.getActiveTimeline().getWriteTimeline().filterPendingCompactionTimeline().getInstants().findAny().isPresent()) {
+        // proceed only if there are no pending compactions
```

Comment:
1. Usually we try to have methods in HoodieActivetimeline that are used by many callers so that we can avoid duplication or errors. For one off cases like this, we keep it away from HoodieActivetimeline. 
2. somehow I feel we are over populating HoodieWriteConfig. This is a very specific use-case and probably we can leave it here. condition here is very specific in the sense that, we need to check if inline is not enabled, but schedule inline is enabled. and I don't see this being used anywhere else.

Reply:
No strong opinions here.